File: app/src/main/java/com/example/roomdatabase/helperclass/AdapterItem.java
package com.example.roomdatabase.helperclass;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.example.roomdatabase.R;

import java.util.ArrayList;

public class AdapterItem extends RecyclerView.Adapter<AdapterItem.PhoneViewHolder> {

    private ArrayList<ItemHelper> phoneLocations;
    private ListItemClickListener listItemClickListener;
    private CheckboxClickListener checkboxClickListener;

    public AdapterItem(ArrayList<ItemHelper> phoneLocations, ListItemClickListener listItemClickListener, CheckboxClickListener checkboxClickListener) {
        this.phoneLocations = phoneLocations;
        this.listItemClickListener = listItemClickListener;
        this.checkboxClickListener = checkboxClickListener;
    }

    public ItemHelper getItem(int position) {
        return phoneLocations.get(position);
    }

    public interface ListItemClickListener {
        void onPhoneListClick(int clickedItemIndex);
    }

    public interface CheckboxClickListener {
        void onCheckboxClick(int position, boolean isChecked);
    }

    public class PhoneViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        public ImageView image;
        public TextView title;
        public RelativeLayout relativeLayout;
        public CheckBox cbHeart;

        public PhoneViewHolder(View itemView) {
            super(itemView);
            image = itemView.findViewById(R.id.imageView);
            title = itemView.findViewById(R.id.text_title);
            relativeLayout = itemView.findViewById(R.id.background_color);
            cbHeart = itemView.findViewById(R.id.cbHeart);

            itemView.setOnClickListener(this);

            cbHeart.setOnCheckedChangeListener((buttonView, isChecked) -> checkboxClickListener.onCheckboxClick(getAdapterPosition(), isChecked));
        }

        @Override
        public void onClick(View v) {
            int clickedPosition = getAdapterPosition();
            listItemClickListener.onPhoneListClick(clickedPosition);
        }
    }

    @Override
    public PhoneViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.activity_pharmacy_list, parent, false);
        return new PhoneViewHolder(view);
    }

    @Override
    public void onBindViewHolder(PhoneViewHolder holder, int position) {
        ItemHelper itemHelper = phoneLocations.get(position);
        holder.image.setImageResource(itemHelper.getImage());
        holder.title.setText(itemHelper.getTitle());
        holder.relativeLayout.setBackgroundColor(itemHelper.getColor());
    }

    @Override
    public int getItemCount() {
        return phoneLocations.size();
    }
}

File: app/src/main/java/com/example/roomdatabase/helperclass/AdapterItem.kt
package com.example.roomdatabase.helperclass

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CheckBox
import android.widget.ImageView
import android.widget.RelativeLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.roomdatabase.R
import java.util.ArrayList

class AdapterItem(
    private val phoneLocations: ArrayList<ItemHelper>,
    private val listItemClickListener: ListItemClickListener,
    private val checkboxClickListener: CheckboxClickListener
) : RecyclerView.Adapter<AdapterItem.PhoneViewHolder>() {

    interface ListItemClickListener {
        fun onPhoneListClick(clickedItemIndex: Int)
    }

    interface CheckboxClickListener {
        fun onCheckboxClick(position: Int, isChecked: Boolean)
    }

    inner class PhoneViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
        var image: ImageView = itemView.findViewById(R.id.imageView)
        var title: TextView = itemView.findViewById(R.id.text_title)
        var relativeLayout: RelativeLayout = itemView.findViewById(R.id.background_color)
        var cbHeart: CheckBox = itemView.findViewById(R.id.cbHeart)

        init {
            itemView.setOnClickListener(this)

            cbHeart.setOnCheckedChangeListener { buttonView, isChecked ->
                checkboxClickListener.onCheckboxClick(adapterPosition, isChecked)
            }
        }

        override fun onClick(v: View) {
            val clickedPosition = adapterPosition
            listItemClickListener.onPhoneListClick(clickedPosition)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PhoneViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.activity_pharmacy_list, parent, false)
        return PhoneViewHolder(view)
    }

    override fun onBindViewHolder(holder: PhoneViewHolder, position: Int) {
        val itemHelper = phoneLocations[position]
        holder.image.setImageResource(itemHelper.image)
        holder.title.text = itemHelper.title
        holder.relativeLayout.setBackgroundColor(itemHelper.color)
    }
    override fun getItemCount(): Int {
        return phoneLocations.size
    }
}
